ts/Core/Series/Series.ts

```diff
@@ -4265,7 +4267,16 @@ class Series {
                  * @name Highcharts.Point#dataGroup
                  * @type {Highcharts.DataGroupingInfoObject|undefined}
                  */
-                point.dataGroup = (series.groupMap as any)[i];
+                if (series && series.groupMap) {
+                    for (j; j < series.groupMap.length; j++) {
+                        if ((series.groupMap[j].start as number) + (series.groupMap[j].length as number) > cropStart) {
+                            groupCropStartIndex = j;
+                            j = series.groupMap?.length;
+                        }
+                    }
+                }
+
```

Comment:
The whole idea of calculating `groupCropStartIndex` was to get the correct `cropStart` (`cropStart` for grouped points, not for the original dataset).

Now, we get `series.cropStart` correct for the grouped points too, so can we reuse it? Something like this:
```js
let groupCropStartIndex: number = options.dataGrouping?.groupAll ? cropStart : 0;
```

Reply:
Or, completely different solution (not touching Highcharts core, only Stock):

Instead of trying to find correct indexes, crop `groupMap` too?  so the only change would be ( in `DataGrouping.ts`):

```js
series.cropStart = croppedData.start; // #15005
```
=>
```js
series.groupMap = series.groupMap.slice(croppedData.start, croppedData.end + 1); // #15005
```

Reply:
I think the `series.groupMap` should remain untouched and inside all groups should exist (even the ones which are currently invisible).
That is why I applied the first suggestion.